testscript/doc.go

```diff
@@ -120,6 +120,35 @@ when testing.Short() is false.
 
 Additional conditions can be added by passing a function to Params.Condition.
 
+An example:
+
+		Condition: func(cond string) (bool, error) {
+			// Assume condition name and args are separated by colon (":")
+			args := strings.Split(cond, ":")
+			// Empty condition is already managed in testscript.run()
+			name := args[0]
+			switch name {
+			case "is_upper":
+				if len(args) < 2 {
+					return false, fmt.Errorf("syntax: [is_upper:word]")
+				}
+				return strings.ToUpper(args[1]) == args[1], nil
```

Comment:
to make the example with an argument a bit more realistic, how about a variant of `exists`? Like the built-in command, but being useful as a condition, like `[exists:foo/bar] mycmd foo/bar`.

```
case "exists":
  _, err := os.Lstat(args[1])
  return err == nil, nil
```

Reply:
Changed the example